src/main/java/org/opensearch/knn/plugin/rest/RestSearchModelHandler.java

```diff
@@ -69,9 +70,22 @@ private void checkUnSupportedParamsExists(RestRequest request) {
         throw new IllegalArgumentException(errorMessage);
     }
 
+    private void searchSizeValidation(RestRequest request) {
+        ValidationException validationException = new ValidationException();
+        if (request.hasParam("size")) {
+            int searchSize = request.paramAsInt("size", 1);
+            if (searchSize > 1000 || searchSize <= 0) {
+                String errorMessage = "Size must be > 0 and <= 1000";
+                validationException.addValidationError(errorMessage);
+                throw validationException;
+            }
+        }
+    }
+
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         checkUnSupportedParamsExists(request);
+        searchSizeValidation(request);
```

Comment:
I will create a new method called validateRequest(request) and move both method call inside that validation method.

Reply:
ack